backend/spec/controllers/spree/admin/products_controller_spec.rb

```diff
@@ -50,6 +50,26 @@
     end
   end
 
+  # regression test for https://github.com/spree/spree/issues/2791
+  context "adding taxons to a product" do
+    let(:product) { create(:product) }
+    let(:first_taxon) { create(:taxon) }
+
+    it "adds a single taxon to a product" do
+      put :update, params: { id: product.to_param, product: { taxon_ids: first_taxon.id.to_s } }
+      expect(flash[:success]).to eq("Product #{product.name.inspect} has been successfully updated!")
+    end
+
+    context "when there are mulitple taxons" do
+      let(:second_taxon) { create(:taxon) }
+
+      it "adds multiple taxons to a product" do
+        put :update, params: { id: product.to_param, product: { taxon_ids: "#{first_taxon.id}, #{second_taxon.id}" } }
+        expect(flash[:success]).to eq("Product #{product.name.inspect} has been successfully updated!")
+      end
+    end
+  end
```

Comment:
You are testing product update, while the PR says product creation. 

Could you add a spec for creation as well?

Reply:
Yep, especially since the before action also applies to create. The code change did expand the behavior to create.

Reply:
Thanks for the feedback, just pushed some more tests!